src/Cli/dotnet/commands/dotnet-workload/search/versions/SearchWorkloadSetsCommand.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Completions;
+using System.Text.Json;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Workloads.Workload.Install;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.TemplateEngine.Cli.Commands;
+
+namespace Microsoft.DotNet.Workloads.Workload.Search.Versions;
+
+internal class SearchWorkloadSetsCommand : WorkloadCommandBase
+{
+    internal readonly int NumberOfWorkloadSetsToTake;
+    private readonly SearchWorkloadSetsFormat _workloadSetOutputFormat;
+    internal readonly IWorkloadManifestInstaller Installer;
+    internal readonly ReleaseVersion SdkVersion;
+    private readonly IWorkloadResolver _workloadResolver;
+
+    private static readonly ManifestId WorkloadPackageIdBase = new("Microsoft.NET.Workloads");
+
+    public SearchWorkloadSetsCommand(ParseResult result, IWorkloadResolverFactory workloadResolverFactory = null, IReporter reporter = null) : base(result, reporter: reporter)
+    {
+        NumberOfWorkloadSetsToTake = result.HasOption(SearchWorkloadSetsParser.TakeOption) ? result.GetValue(SearchWorkloadSetsParser.TakeOption) : 5;
+        _workloadSetOutputFormat = result.GetValue(SearchWorkloadSetsParser.FormatOption);
+        workloadResolverFactory = workloadResolverFactory ?? new WorkloadResolverFactory();
+        var creationResult = workloadResolverFactory.Create();
+
+        SdkVersion = creationResult.SdkVersion;
+        _workloadResolver = creationResult.WorkloadResolver;
+
+        Installer = WorkloadInstallerFactory.GetWorkloadInstaller(
+                reporter,
+                new SdkFeatureBand(SdkVersion),
+                _workloadResolver,
+                Verbosity,
+                creationResult.UserProfileDir,
+                !SignCheck.IsDotNetSigned(),
+                restoreActionConfig: new RestoreActionConfig(result.HasOption(SharedOptions.InteractiveOption)),
+                elevationRequired: false,
+                shouldLog: false);
+    }
+
+    protected override void ShowHelpOrErrorIfAppropriate(ParseResult parseResult)
+    {
+
+    }
+
```

Comment:
This is an odd call-out - the `CommandBase` that `WorkloadCommandBase` inherits from unconditionally calls the 'bail out if necessary' extension method we have, but I think that's bad behavior. In the tab-completion case, you very well may get invalid/erroring parse results, and so throwing those errors in the completion context prevents completions from being shown.